lib/pipeline.js

```diff
@@ -355,6 +380,158 @@ class PipelineModel extends BaseModel {
             .then(() => this);
     }
 
+    /**
+     * Checks if any admin from this.admins has github permission to given scmUri
+     * @method _hasAdminPermission
+     * @param  {String}    scmUri
+     * @return {Promise}
+     */
+    async _hasAdminPermission(scmUri) {
+        /* eslint-disable no-restricted-syntax */
+        for (const username of Object.keys(this.admins)) {
+            // eslint-disable-next-line global-require
+            const UserFactory = require('./userFactory');
+            const factory = UserFactory.getInstance();
+
+            // eslint-disable-next-line no-await-in-loop
+            const user = await factory.get({
+                username,
+                scmContext: this.scmContext
+            });
+
+            // eslint-disable-next-line no-await-in-loop
+            const permission = await user.getPermissions(scmUri);
+
+            if (permission.admin) {
+                return true;
+            }
+        }
+        /* eslint-enable no-restricted-syntax */
+
+        return false;
+    }
+
+    /**
+     * Get a pipeline given a scmUrl
+     * @method getPipeline
+     * @param {Array} scmUrl  checkout url for a repository
+     * @return {Promise}
+     */
+    async _getScmUri(scmUrl) {
+        // eslint-disable-next-line global-require
+        const PipelineFactory = require('./pipelineFactory');
+        const pipelineFactory = PipelineFactory.getInstance();
+        const token = await this.token;
+        const admin = await this.admin;
+        const scmContext = admin.scmContext;
+        const scmUri = await pipelineFactory.scm.parseUrl({
+            scmContext,
+            checkoutUrl: scmUrl,
+            token
+        });
+
+        return scmUri;
+    }
+
+    /**
+     * Update or create a pipeline given a scmUrl
+     * @method createOrUpdatePipeine
+     * @param {Array} scmUrl  checkout url for a repository
+     * @return {Promise}
+     */
+    async _createOrUpdatePipeine(scmUrl) {
+        // eslint-disable-next-line global-require
+        const PipelineFactory = require('./pipelineFactory');
+        const pipelineFactory = PipelineFactory.getInstance();
+        const admins = this.admins;
+        const scmContext = this.scmContext;
+        const scmUri = await this._getScmUri(scmUrl);
+        const hasAdminPermission = await this._hasAdminPermission(scmUri);
+
+        if (!hasAdminPermission) {
+            // need to figure out how to bubble up this err to user
+            // eslint-disable-next-line no-console
+            console.log(`Admins of pipeline:${this.id} has no admin permission to ${scmUrl}`);
```

Comment:
Perhaps this can be worded like:

"No admins of pipeline:X have admin permissions on scmUrl"